lib/value/src/kind/collection.rs

```diff
@@ -276,6 +276,57 @@ impl<T: Ord> From<BTreeMap<T, Kind>> for Collection<T> {
     }
 }
 
+impl std::fmt::Display for Collection<Field> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if self.unknown.is_some() || self.known.is_empty() {
+            // Simple representation, we can improve upon this in the future.
+            return f.write_str("object");
+        }
+
+        f.write_str("{ ")?;
+
+        let mut known = self.known.iter().peekable();
+        while let Some((key, kind)) = known.next() {
+            write!(f, "{key}: {kind}")?;
+            if known.peek().is_some() {
+                f.write_str(", ")?;
+            }
+        }
+
+        f.write_str(" }")?;
+
+        Ok(())
+    }
+}
+
+impl std::fmt::Display for Collection<Index> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if self.unknown.is_some() || self.known.is_empty() {
+            // Simple representation, we can improve upon this in the future.
+            return f.write_str("array");
+        }
```

Comment:
This could potentially get confusing. I think it would be worth adding at minimum an extra note if there are some defined indexes, something like:

```rust
if !self.known.is_empty() {
  return f.write_str("array + more"); // I'm not sure exactly the right thing to say here.
} else {
  return f.write_str("array");
}
```

Reply:
It's no different from the current situation, so I'd like to keep that for a future improvement.